src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,183 @@
+use std::iter;
+use std::io::{Result, Write};
+
+use env::Env;
+use file_saver::save_to_file;
+use library::{self, Library, ToStripperType};
+use library::Type as LType;
+
+use stripper_interface::{FILE, FILE_COMMENT, MOD_COMMENT, TypeStruct};
+
+const TAB_SIZE : usize = 4;
+
+pub fn generate(env: &Env) {
+    let path =  env.config.target_path.join("doc.cmts");
+    println!("Generating documentation {:?}", path);
+    save_to_file(&path, env.config.make_backup,
+        |w| generate_doc(w, &env.library));
+}
+
+fn generate_doc(w: &mut Write, lib: &Library) -> Result<()> {
+    for i in 0..lib.namespaces.len() {
+        let mut output = String::new();
+        let namespace = &lib.namespaces[i];
+
+        if let Some(ref namespace_doc) = namespace.doc {
+            for line in namespace_doc.split("\n") {
+                output.push_str(&format!("{}{}\n", FILE_COMMENT, line));
+            }
+        }
+        try!(create_sub_docs(&mut output, &namespace.constants, None));
+        for i in 0..namespace.types.len() {
```

Comment:
Oh the `.as_ref`, I totally forgot it... And yes, it's because of the moved errors that came back to this syntax.